src/LesserPhp/CompilerInterface.php

```diff
@@ -0,0 +1,246 @@
+    public function checkedCompile($in, $out);
+
+    /**
+     * Execute lessphp on a .less file or a lessphp cache structure
+     *
+     * The lessphp cache structure contains information about a specific
+     * less file having been parsed. It can be used as a hint for future
+     * calls to determine whether or not a rebuild is required.
+     *
+     * The cache structure contains two important keys that may be used
+     * externally:
+     *
+     * compiled: The final compiled CSS
+     * updated: The time (in seconds) the CSS was last compiled
+     *
+     * The cache structure is a plain-ol' PHP associative array and can
+     * be serialized and unserialized without a hitch.
+     *
+     * @param mixed $in    Input
+     * @param bool  $force Force rebuild?
+     *
+     * @return array lessphp cache structure
+     * @throws \LesserPhp\Exception\GeneralException
+     */
+    public function cachedCompile($in, $force = false);
+
+    /**
+     * parse and compile buffer
+     * This is deprecated
+     *
+     * @param null $str
+     * @param null $initialVariables
+     *
+     * @return int|string
+     * @throws \LesserPhp\Exception\GeneralException
+     * @deprecated
+     */
+    public function parse($str = null, $initialVariables = null);
+
+    /**
+     * @param string $name
+     */
+    public function setFormatter($name);
+
+    public function setFormatterClass($formatter);
+
+    /**
+     * @param bool $preserve
+     */
+    public function setPreserveComments($preserve);
+
+    /**
+     * @param string   $name
+     * @param callable $func
+     */
+    public function registerFunction($name, callable $func);
+
+    /**
+     * @param string $name
+     */
+    public function unregisterFunction($name);
+
+    /**
+     * @param array $variables
+     */
+    public function setVariables(array $variables);
+
+    /**
+     * @param $name
+     */
+    public function unsetVariable($name);
+
+    /**
+     * @param string[] $dirs
+     */
+    public function setImportDirs(array $dirs);
+
+    /**
+     * @param string $dir
+     */
+    public function addImportDir($dir);
+
+    /**
+     * @return string[]
+     */
+    public function getImportDirs();
+
+    /**
+     * @param string $file
+     */
+    public function addParsedFile($file);
+
+    /**
+     * Uses the current value of $this->count to show line and line number
+     *
+     * @param string $msg
+     *
+     * @throws GeneralException
+     */
+    public function throwError($msg = null);
+
+    /**
+     * prefix of abstract properties
+     *
+     * @return string
+     */
+    public function getVPrefix();
+
+    /**
+     * prefix of abstract blocks
+     *
+     * @return string
+     */
+    public function getMPrefix();
+
+    /**
+     * @return string
+     */
+    public function getParentSelector();
+
+    public function setImportDisabled();
+
+    /**
+     * @return bool
+     */
+    public function isImportDisabled();
+}
```

Comment:
I do not know what you mean exactly (can't find it).

This interface is not actually used by the Compiler it's just there to help find public methods. Maybe I should not have included it.